source/scpp/extra/DSizeChecks.cpp

```diff
@@ -46,7 +46,7 @@ CPPSIZEOF(CryptoKeyType)
 CPPSIZEOF(PublicKeyType)
 CPPSIZEOF(SignerKeyType)
 CPPSIZEOF(PublicKey)
-CPPSIZEOF(Signature)
+// Signature is removed because it is the same as Hash.
```

Comment:
Does the test cause problem ?

Reply:
```

Reply:
I changed the pointer of SCPEnvelope to non-pointer.

Reply:
Could you check how it is?

Reply:
There was no warning. But I think it's better to change the same as other structs

Reply:
Could we add something like a static assert in the C++ code?

```C++
#include <type_traits> 

static_assert(std::is_same<Signature, Hash>::value, "Signature and Hash must be the same type")
```